### PR TITLE
Refactored login

### PR DESCRIPTION
**Pull request by smabhena**

### CODE REVIEW

File: Mooov/Mooov/Log in/LogInViewController.swift [hunk 1]
```diff
@@ -12,19 +12,28 @@ class LogInViewController: UIViewController {
...
-            
-            if let navigator = self.navigationController {
-                navigator.pushViewController(tabBarController, animated: true)
-                navigator.setNavigationBarHidden(true, animated: false)
-            }
+        guard let username = usernameField.text else { return }
+        guard let password = passwordField.text else { return }
+        
+        if viewModel.logInUser(username, password) {
+            navigateToHomePage()
```
Comment (bgouws) [inline Mooov/Mooov/Log in/LogInViewController.swift:36]: Why not have your ViewModel decided what should happen with a delegate? 

The thought process being: call the `ViewModel` to log the user in and then using a delegate function based on if the login was successful to either navigate the user to the next page or display an error message letting the user know that the login was unsuccessful.

**Review by bgouws — CHANGES_REQUESTED** [review]
Just one more enhancement 😄

**Review by kirsty14 — APPROVED** [review]
Looking good. From my perspective you have successfully refactored to MVVM

File: Mooov/Mooov/Log in/LogInViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+protocol LogInViewModelDelegate: AnyObject {
+    func navigateToHomePage()
+    func showError(error: String)
+}
+
+class LogInViewModel {
+    private weak var delegate: LogInViewModelDelegate?
+    
+    init(
```
Comment (bgouws) [inline Mooov/Mooov/Log in/LogInViewModel.swift:18]: Just some alignment that needs to be fixed here: 

`init(delegate: LogInViewModelDelegate) {`

**Review by bgouws — CHANGES_REQUESTED** [review]

**Review by bgouws — APPROVED** [review]

**Review by kirsty14 — APPROVED** [review]
Looking good 👍🏻

**Pull request merged**